src/main/java/hudson/plugins/scm/koji/client/KojiBuildDownloader.java

```diff
@@ -99,7 +97,28 @@ public KojiBuildDownloadResult invoke(File workspace, VirtualChannel channel) th
             targetDir.mkdirs();
         }
         List<String> rpmFiles = downloadRPMs(targetDir, build);
-        
+        String srcUrl = "";
+        for (String suffix : RPM.Suffix.INSTANCE.getSuffixes()) {
```

Comment:
This loop sets last reachable url to `build`. Is it correct behavior?

Reply:
It can be optimized when `getSuffixes()` is reversed and take first reachable url. You avoid to testing url reachability for urls that you might throw away later.